screens/ReportsScreen.js

```diff
@@ -19,14 +26,48 @@ class ReportsScreen extends React.Component {
 
   constructor(props, context) {
     super(props, context)
+
+    this.state = {
+      userScope: false
+    }
+  }
+
+  async componentDidMount() {
+    // await this.handleUserScope()
+    await this.handleCurrentUserRole()
+  }
+  componentDidUpdate(prevProps, prevState) {
+    if ((prevProps?.currentUser?.username !== this.props?.currentUser?.username || prevProps?.client?.clientName !== this.props?.client?.clientName)) {
+
+      this.handleCurrentUserRole()
+      // this.handleUserScope()
+    }
+  }
+
+  // tmp method: user roles
+  handleCurrentUserRole = async () => {
+    let token = await getToken()
+    this.props?.getCurrentUser(token.username)
+    let permission = getCurrentUserRole(this.props?.currentUser)
+
+    console.log("role permission = ", permission)
+    this.setState({userScope: permission})
```

Comment:
The retrieval of client user's permission should be relocated to PermissionCheckTouchable class.